routers/repo/issue.go

```diff
@@ -281,9 +283,61 @@ func RetrieveRepoMetas(ctx *middleware.Context, repo *models.Repository) []*mode
 	return labels
 }
 
+func getFileContentFromDefaultBranch(ctx *middleware.Context, filename string) (string, bool) {
+	var r io.Reader
+	var bytes []byte
+
+	if ctx.Repo.Commit == nil {
+		var err error
+		ctx.Repo.Commit, err = ctx.Repo.GitRepo.GetBranchCommit(ctx.Repo.Repository.DefaultBranch)
+		if err != nil {
+			return "", false
+		}
+	}
+
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(filename)
+	if err != nil {
+		return "", false
+	}
+	r, err = entry.Blob().Data()
+	if err != nil {
+		return "", false
+	}
+	bytes, err = ioutil.ReadAll(r)
+	if err != nil {
+		return "", false
+	}
+	return string(bytes), true
+}
+
+func setIssueTemplateIfExists(ctx *middleware.Context) {
+	possibleFiles := []string{"ISSUE_TEMPLATE.md", ".gogs/ISSUE_TEMPLATE.md", ".github/ISSUE_TEMPLATE.md"}
+
+	for _, filename := range possibleFiles {
+		content, found := getFileContentFromDefaultBranch(ctx, filename)
+		if found {
+			ctx.Data["IssueTemplate"] = content
+			return
+		}
+	}
+}
+
+func setPullRequestTemplateIfExists(ctx *middleware.Context) {
```

Comment:
`setIssueTemplateIfExists` and `setPullRequestTemplateIfExists` can be abstracted again.